tests/unit/sys/test_options.cxx

```diff
@@ -65,6 +65,28 @@ TEST_F(OptionsTest, IsSectionNotCaseSensitive) {
   ASSERT_TRUE(options.isSection("Subsection"));
 }
 
+TEST_F(OptionsTest, CompoundName) {
+  Options options;
+
+  // make sure options is initialized as a section
+  options["testkey"] = 1.;
+
+  ASSERT_TRUE(options.isSection());
+  ASSERT_FALSE(options["testkey"].isSection());
+  ASSERT_TRUE(options.isSection(""));
+  ASSERT_FALSE(options.isSection("subsection"));
+
+  options["subsection:testkey"] = 1.;
```

Comment:
Please could you change "testkey" and `1.` here? I get nervous when test values could potentially clash!